src/v2/components/ChannelContents/index.tsx

```diff
@@ -1,124 +1,87 @@
+import { getConnectableType } from './lib/getConnectableType'
+import connectableBlokk from './queries/connectableBlokk'
+
 const SortableGrid = SortableContainer(({ onSortEnd: _onSortEnd, ...rest }) => (
   <Grid {...rest} />
 ))
 
 interface Props {
-  chunkSize?: number
   channel: ChannelContentsData
   pusherChannel?: any
   socket?: any
 }
 
-interface ChannelContentsProps extends Props {
-  moveConnectable: (props: any) => Promise<any>
-  client: ApolloClient<any>
-}
-
-const ChannelContents: React.FC<ChannelContentsProps> = memo(
-  ({
-    chunkSize = 10,
-    channel,
-    client,
-    moveConnectable,
-    pusherChannel,
-    socket,
-    ...rest
-  }) => {
-    // Used to load/unload waypoints
-    const [activeQueries, setActiveQueries] = useState<
-      ActiveQueriesCollection.ActiveQueries
-    >({})
-
-    // Handles ordering of block grid items
-    const [connectables, setConnectables] = useState(channel.skeleton)
-
-    // Handles actual contents of block grid items
-    const [collection, setCollection] = useState(
-      ConnectableCellsCollection.normalize(channel.initial_contents)
+const ChannelContents: React.FC<Props> = memo(
+  ({ channel, pusherChannel, socket, ...rest }) => {
+    const {
+      blocks,
+      getPage,
+      getPageFromIndex,
+      hasQueriedPage,
+      moveBlock,
+      removeBlock,
+      addBlock,
+      contentCount,
+    } = usePaginatedBlocks({
+      channelId: channel.id,
+    })
+
+    const onItemIntersected = useCallback(
+      (index: number) => {
+        const page = getPageFromIndex(index)
+        if (!hasQueriedPage(page)) {
+          getPage(page)
+        }
+      },
+      [getPage, getPageFromIndex, hasQueriedPage]
     )
 
-    const addConnectable = useCallback(newConnectable => {
-      setConnectables(prevConnectables => {
-        if (
-          !prevConnectables.some(
-            c =>
-              c.id === newConnectable.id &&
-              c.type.toUpperCase() === newConnectable.type.toUpperCase()
-          )
-        ) {
-          return [
-            {
-              id: newConnectable.id,
-              type: { BLOCK: 'Block', CHANNEL: 'Channel' }[
-                newConnectable.type.toUpperCase()
-              ],
-              __typename: 'SkeletalConnectable',
-            },
-            ...prevConnectables,
-          ]
-        }
+    const parsePayload = useCallback(
+      ({ id }) => ({
+        id: id.toString(),
+      }),
+      []
+    )
 
-        return prevConnectables
-      })
-    }, [])
+    const client = useApolloClient()
 
     const updateConnectable = useCallback(
-      connectable => {
-        loadSkeleton({
-          client,
-          channelId: channel.id,
-          pageSkeleton: [connectable],
-          collection,
-          queryOptions: {
-            fetchPolicy: 'network-only',
+      ({ id }: { id: string | number }) => {
+        client.query<ConnectableBlokk, ConnectableBlokkVariables>({
```

Comment:
Because channel content is stored in apollo cache, all we need to do to revalidate a block is to query for that same block again! Then apollo will automatically update the cached data _and_ re render any active queries that are referencing the value